sycl/source/detail/os_util.cpp

```diff
@@ -254,32 +231,12 @@ void OSUtil::alignedFree(void *Ptr) {
 // Make all directories on the path, throws on error.
 int OSUtil::makeDir(const char *Dir) {
   assert((Dir != nullptr) && "Passed null-pointer as directory name.");
-  if (isPathPresent(Dir))
-    return 0;
-
-// older GCC doesn't have full C++ 17 support.
```

Comment:
is 
"#if defined(__has_include) && __has_include(<filesystem>)" really false for old gcc?
would it be better to use feature test macro? https://en.cppreference.com/w/cpp/feature_test#cpp_lib_filesystem